app/services/users/omniauth.rb

```diff
@@ -0,0 +1,37 @@
+class Users::Omniauth
+  attr_reader :auth
+
+  def initialize(auth)
+    @auth = auth
+  end
+
+  def get
+    user
+  end
+
+  private
+
+  delegate :provider, :uid, :info, to: :auth
+  delegate :email, to: :info
+
+  def user
+    user = find_user
+    user ||= create_user
```

Comment:
@sinpai isn't 
```ruby
user = find_user
user ||= create_user
user
```

the same as
```ruby
find_user || create_user
```

?

Reply:
@sinpai we can go even further - memoize `user` variable
```ruby
def user
  @_user ||= find_user || create_user
end
```
in order to have ability to reuse `user` method